### PR TITLE
TinyUSB stack ported to MCXA and MXCN freedom boards. 

### PR DESCRIPTION
**Pull request by jerpa77**
TinyUSB stack ported to MCX A and MXC N freedom boards. 

**Notes**
1. MCX A and MCX N families created under hw/bsp folder. Due to differing compiler options required by different versions of ARM Cortex M-33 for MCX A and MCX N, they were not grouped together in the same family.

2. Build code for make and cmake tools added under example/build_system to support the ARM Cortex M-33 core with no DSP and no FPU. 

3. Since the NXP SDK repository is not up to date yet, the tools/get_deps.py file was modified to get the dependencies for MCX A, MCX N, etc. from our fork https://github.com/wavenumber-eng/mcux-sdk. We manually added the data from the MCUXpresso SDK generator to  support MCX A. In a few weeks NXP will update their SDK repository, so no github fork will be necessary.

4. The building code was implemented and tested to work with make and cmake. The following examples have been tested on the MCX A and MCX N boards:
- board_test
- cdc_dual_ports
- cdc_msc
- hid_composite

5. USB Compliance Verification successfully passed the following tests:
- Chapter 9 Tests [USB 2 devices]
- HID Tests
- MSC Tests

### CODE REVIEW

File: src/portable/chipidea/ci_fs/ci_fs_mcxa.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * This file is part of the TinyUSB stack.
+ */
+
+#ifndef _CI_FS_MCXA_H
```
Comment (hathach) [inline src/portable/chipidea/ci_fs/ci_fs_mcxa.h:27]: Plesae remove this file, and use the existing mcx.h, we can sort out the mcxa vs mcxn there using existing macros.

File: README.rst [hunk 2]
```diff
@@ -139,7 +139,9 @@ Following CPUs are supported, check out `Supported Devices`_ for comprehensive l
 |              +---------+--------------------------------------------------+
 |              | LPC     | 11u, 13, 15, 17, 18, 40, 43, 51u, 54, 55         |
 |              +---------+--------------------------------------------------+
-|              | MCX     | N9                                               |
+|              | MCXA    | A153                                             |
```
Comment (hathach) [inline README.rst:142]: please have this as 

| MCX | A15, N9 |

**Review by hathach — COMMENTED** [review]
Thank you very much for your PR. It looks great, though there is some change request below:

> MCX A and MCX N families created under hw/bsp folder. Due to differing compiler options required by different versions of ARM Cortex M-33 for MCX A and MCX N, they were not grouped together in the same family.

besides the DSP/FPU option, is there any other difference, we can solve this by either using the MCU_VARIANT check in family.mk/cmake or just set the correct `CPU_CORE` in the board.mk/cmake instead. That would both A & N series under one mcx directory

**Comment by jerpa77** [comment]
Thank you a lot for your feedback.
I already made the pull request for your mcux-sdk fork.
When we have the mcux-sdk updated I'm going to push the corrections.

**Comment by jerpa77** [comment]
Hello Thach! 
I already made the changes

**Comment by hathach** [comment]
I try to resolve conflict with master, and also want to make changes to the PR. Please edit PR to "**allow maintainer to edit this pull request**"

**Comment by jerpa77** [comment]
Hi @hathach 
I read that this option is only for user-owned repositories, but I sent you an invitation to have permissions on the repository.

**Comment by hathach** [comment]
resolved the conflict, I don't why, but your PR contains also lots of unrelated changes to other files such as cdc_msc_freertos, nrf etc.. probably the git branch is messed up somehow. I have reverted those changes.

**Review by hathach — APPROVED** [review]
thank you for your PR. Everything work perfecty now, I made some changes to get it built with all examples. Currenlty ci failed on hil test, but that is a totally different issue by local pi4 setup. Will fix  that in master.

**Pull request merged**